### PR TITLE
app: standardize initialization

### PR DESCRIPTION
**Pull request by Garfield-fr**
Initializes the application according to angular standards.

* Adds a service for initialization.
* Adds a service for router initialization.
* Improves app component template.
* Improves tests.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/app.component.ts [hunk 1]
```diff
@@ -15,134 +15,46 @@
...
-    private localStorageService: LocalStorageService,
-    private librarySwitchService: LibrarySwitchService,
-    private router: Router,
-    private organisationService: OrganisationService
-    ) {
-      this.initializeEvents();
-    }
+    private _userService: UserService,
+    private _appRouterEventService: AppRouterEventService,
+    private _librarySwitchService: LibrarySwitchService
```
Comment (zannkukai) [inline projects/admin/src/app/app.component.ts:48]: missing docString for this variable.

**Review by zannkukai — APPROVED** [review]
Write a better commit message to avoid a message from @iGormilhit :-)

**Comment by Garfield-fr** [comment]
> Write a better commit message to avoid a message from @iGormilhit :-)

done

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message changes:
```
app: standardize initialization

Initializes the application according to angular standards.

* Adds a service for initialization.
* Adds a service for router initialization.
* Improves app component template.
```
Title too long, missing uppercase initials of the item list.

**Comment by Garfield-fr** [comment]
> Commit message changes:
> 
> ```
> app: standardize initialization
> 
> Initializes the application according to angular standards.
> 
> * Adds a service for initialization.
> * Adds a service for router initialization.
> * Improves app component template.
> ```
> 
> Title too long, missing uppercase initials of the item list.

done. Thank you for the message proposal.

**Review by iGormilhit — APPROVED** [review]

File: projects/admin/src/app/app.component.html [hunk 2]
```diff
@@ -15,12 +15,14 @@
   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 <ngx-spinner type="ball-spin-clockwise" size="medium"></ngx-spinner>
-<div *ngIf="!user" translate>Loading…</div>
-<ng-container *ngIf="user">
+<ng-template #usernotloaded>
+  <div translate>Loading…</div>
+</ng-template>
```
Comment (sebastiendeleze) [inline projects/admin/src/app/app.component.html:20]: Not sure, but I think ngx-spinner can handle a message like this.

**Review by sebastiendeleze — APPROVED** [review]

File: projects/admin/src/app/app.component.html (hunk 2 above)
Comment (Garfield-fr) [reply projects/admin/src/app/app.component.html]: done

File: projects/admin/src/app/service/app-init.service.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    private _userService: UserService,
+    private _appConfigService: AppConfigService,
+    private _translateService: TranslateService,
+    private _organisationService: OrganisationService,
+    private _localeStorageService: LocalStorageService
+  ) { }
+
+  /** Application load */
+  load() {
+    return new Promise((resolve) => {
```
Comment (jma) [inline projects/admin/src/app/service/app-init.service.ts:47]: Can you use an Observable here to be consistent with the rest of the application? Else ignore my comment.

File: projects/admin/src/app/service/app-init.service.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+   */
+  constructor(
+    private _userService: UserService,
+    private _appConfigService: AppConfigService,
+    private _translateService: TranslateService,
+    private _organisationService: OrganisationService,
+    private _localeStorageService: LocalStorageService
+  ) { }
+
+  /** Application load */
```
Comment (jma) [inline projects/admin/src/app/service/app-init.service.ts:45]: return value is missing in the comment.

File: projects/admin/src/app/service/app-router-event.service.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    private _localeStorage: LocalStorageService,
+    private _appConfigService: AppConfigService
+  ) { }
+
+  /** Initialize event */
+  initializeEvents() {
+    this.eventNavigationStart();
+  }
+
+  /** Event navigation start */
```
Comment (jma) [inline projects/admin/src/app/service/app-router-event.service.ts:46]: A small comment of what is done here is welcome.

**Review by jma — APPROVED** [review]

File: projects/admin/src/app/service/app-init.service.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/admin/src/app/service/app-init.service.ts:47]: Remove Promise (not necessary)

File: projects/admin/src/app/service/app-init.service.ts (hunk 4 above)
Comment (Garfield-fr) [reply projects/admin/src/app/service/app-init.service.ts:45]: done

File: projects/admin/src/app/service/app-router-event.service.ts (hunk 5 above)
Comment (Garfield-fr) [reply projects/admin/src/app/service/app-router-event.service.ts:46]: done

**Pull request closed without merging**